usr/src/lib/libdisasm/common/dis_arm_64.c

```diff
@@ -2016,19 +2041,27 @@ a64_dis_dataproc(dis_handle_t *dhp, uint32_t in, char *buf, size_t buflen)
 			    ", #0x%" PRIx64 ", lsl #12", imm);
 			break;
 		}
-		if ((dop->flags & DPI_ADRP_F) != 0) {
+		if ((dop->flags & DPI_ADRP_F) != 0 &&
+		    dpi.rd.width == A64_REGWIDTH_64) {
 			dis_handle_arm_64_t *dhx = dhp->dh_arch_private;
+			uint_t i = dpi.rd.id;
 
-			if (dhx->dha_adrp_imm != 0 &&
-			    dpi.rd.id == dhx->dha_adrp_reg.id &&
-			    dpi.rd.id == dpi.rn.id &&
-			    dhp->dh_addr == dhx->dha_adrp_addr +
-			    sizeof (uint32_t)) {
+			/*
```

Comment:
This is what we do instead of flow analysis on the registers, right?
Does a 10 instruction window seem good in your comparisons?  I don't think misfiring if the `adrp` destination is clobbered is a real problem in practice, and the symbol annotations are so useful.

Reply:
Almost all of them have the `adrp` and `add` consecutively, in a few cases there were between 1-3 instructions between them, it usually looks like a few `adrl`s are interleaved, as in `adrp`, `adrp`, `adrp`, `add`, `add`, `add`. I didn't see anything higher than 3 in my testing - 10 is arbitrary but felt big enough to catch these. There's probably no harm in removing that limit though. Happy to do that if you agree.

Reply:
No, by the sound of it that's a great default.  We can always tweak it, I'm just glad to  know it catches most